keps/sig-cli/kustomize-secret-generator-plugins.md

```diff
@@ -205,52 +266,114 @@ are out of scope for this KEP, however this KEP's
 implementation will make it much easier to create other
 kinds of plugins.
 
-Go Plugins that become popular have a clear
-path to becoming a builting - so if someone writes, say,
-a general Exec plugin, we can easily promote it to a
-builtin (by virtue of the fact that it's written in Go,
-and because of the choices made in this KEP for
-describing a plugin in the kustomization file).
+### goplugin limitations
+
+The first plugin mechanism will be goplugins, since by
+design they are trivial to make for a Go program (like
+kustomize).  However, they have limitations, and are
+to some  extent an experimental feature of Go tooling.
+
+#### Not shareable as object code
+
+The shared object (`.so`) files created via `go build
+-buildmode=plugin` cannot be reliably used by a loading
+program unless both the program and the plugin were compiled
+with the same version of Go, with the same libs, on the same
+machine (see, e.g. [this golang issue](https://github.com/golang/go/issues/18827)).
+
+Therefore, the Go developer who elects to write their
+key:value generator as a goplugin is obligated to compile a
+new kustomize binary as well as the plugin.
```

Comment:
must create a new distribution by building both kustomize and the plugin and packaging them together.

Reply:
done